src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java

```diff
@@ -863,47 +871,24 @@ public String delete(String indexName, String type, String id) {
 				: getPersistentEntityFor(clazz).getIndexName();
 		String typeName = hasText(deleteQuery.getType()) ? deleteQuery.getType()
 				: getPersistentEntityFor(clazz).getIndexType();
-		Integer pageSize = deleteQuery.getPageSize() != null ? deleteQuery.getPageSize() : 1000;
-		Long scrollTimeInMillis = deleteQuery.getScrollTimeInMillis() != null ? deleteQuery.getScrollTimeInMillis()
-				: 10000l;
 
-		SearchQuery searchQuery = new NativeSearchQueryBuilder().withQuery(deleteQuery.getQuery()).withIndices(indexName)
-				.withTypes(typeName).withPageable(PageRequest.of(0, pageSize)).build();
+		DeleteByQueryRequest deleteByQueryRequest = new DeleteByQueryRequest(indexName) //
+				.setDocTypes(typeName) //
+				.setQuery(deleteQuery.getQuery()) //
+				.setAbortOnVersionConflict(false) //
+				.setRefresh(true);
 
-		SearchResultMapper deleteEntryResultMapper = new SearchResultMapperAdapter() {
+		if (deleteQuery.getPageSize() != null)
+			deleteByQueryRequest.setBatchSize(deleteQuery.getPageSize());
 
-			@Override
-			public <T> AggregatedPage<T> mapResults(SearchResponse response, Class<T> clazz, Pageable pageable) {
-				return new AggregatedPageImpl<>((List<T>) Arrays.asList(response.getHits().getHits()), response.getScrollId());
-			}
-		};
-
-		ScrolledPage<SearchHit> scrolledResult = startScroll(scrollTimeInMillis, searchQuery, SearchHit.class,
-				deleteEntryResultMapper);
-		BulkRequest request = new BulkRequest();
-		List<SearchHit> documentsToDelete = new ArrayList<>();
-
-		do {
-			documentsToDelete.addAll(scrolledResult.getContent());
-			scrolledResult = continueScroll(scrolledResult.getScrollId(), scrollTimeInMillis,
-					SearchHit.class, deleteEntryResultMapper);
-		} while (scrolledResult.getContent().size() != 0);
-
-		for (SearchHit entry : documentsToDelete) {
-			request.add(new DeleteRequest(entry.getIndex(), typeName, entry.getId()));
-		}
+		if (deleteQuery.getScrollTimeInMillis() != null)
+			deleteByQueryRequest.setScroll(TimeValue.timeValueMillis(deleteQuery.getScrollTimeInMillis()));
 
-		if (request.numberOfActions() > 0) {
-			BulkResponse response;
-			try {
-				response = client.bulk(request);
-				checkForBulkUpdateFailure(response);
-			} catch (IOException e) {
-				throw new ElasticsearchException("Error while deleting bulk: " + request.toString(), e);
-			}
+		try {
+			client.deleteByQuery(deleteByQueryRequest, RequestOptions.DEFAULT);
+		} catch (IOException e) {
+			throw new ElasticsearchException("Error for search request with scroll: " + deleteByQueryRequest.toString(), e);
```

Comment:
please adjust the error message to the fact that we are deleting and not searching

Reply:
Thanks, will change the message